mlflow/pipelines/pipeline.py

```diff
@@ -152,6 +152,35 @@ def _get_step(self, step_name) -> BaseStep:
             )
         return self._steps[step_names.index(step_name)]
 
+    def _get_default_step(self) -> BaseStep:
+        steps = self._steps or self._resolve_pipeline_steps()
```

Comment:
Can we remove the `or` and call `self._resolve_pipeline_steps()`?

```suggestion
        steps = self._resolve_pipeline_steps()
```